src/test/scala/scala/compat/java8/StepConvertersTest.scala

```diff
@@ -438,6 +438,23 @@ class StepConvertersTest {
   }
 
   @Test
+  def shortWidening() {
+    implicit val spec = SpecCheck(_.isInstanceOf[IntStepper])
+
+    good( Array[Short](654321.toShort).stepper )
+
+    //TODO: None of these currently work because there are no native Stepper implementations. This does not only
+    // affect widening conversions though. While you can get, for example, an IntStepper for a WrappedArray[Int],
+    // all values have to go through a boxing/unboxing step!
```

Comment:
@Ichoran, I think this part brings us back to the role of Steppers in the API and whether they should be exposed to the user. If so, we may want to provide widening conversions at the Stepper level for convenience, otherwise we could probably remove many of the existing primitive Stepper implementations which are based on boxed collections, and apply an unboxing Stepper similar to the new widening stepper adapters.